policies/coding-style.md

```diff
@@ -591,21 +591,89 @@ should be used. This is more commonly referred to as C90. ISO/IEC 9899:1999
 (also known as C99) is not supported on some platforms that OpenSSL is
 used on and therefore should be avoided.
 
-## Chapter 15: Miscellaneous
+## Chapter 15: Expressions
 
-Do not use `!` to check if a pointer is NULL, or to see if a str...cmp
-function found a match.  For example, these are wrong:
+Avoid needless parentheses.
+For example, do not write
+```c
+    if ((p == NULL) && (!f(((2 * x) + y) == (z++))))
+```
+but
+```c
+    if (p == NULL && !f(2 * x + y == z++)).
+```
+
+For clarity, always put parentheses when mixing `&&` and `||` operations.
```

Comment:
I suspect we will want a whole lot more exceptions.  C has lots of operators and almost nobody learns the precedence order.  Perhaps this is a _common sense_ rule, reduce parenthesis as far as is reasonable but no further?

E.g. `a <= b == c >= d + 1` is a perfectly well defined C expression with minimal parenthesis but a lot of folks won't instantly know what it's doing.

Reply:
I've extended the text and the examples this way, also regarding the so-called bitwise operators `&`, `^`, and `|`.

Reply:
The whole approach of preventing "extra" parentheses is something I think should be removed completely - as mistakes do happen and parentheses do significantly improve readability.

Reply:
@t-j-h I wholeheartedly disagree. For me a code with needless parentheses is much less readable actually. I always have to count the parentheses and so on to understand what is going on there. The example in this PR is a perfect one for me. The condition with the extra parentheses looks like parentheses hell to me where the one without is perfectly readable.

Reply:
In most real life cases I'd agree with @t-j-h. I never remember the operators priority :(

Reply:
Well, what to consider a reasonable amount of extra parentheses is a matter of both taste and fluency with C.
This could be debated ad nauseam.

Reply:
If this ever becomes a problem in practice (in a code review) just update the coding style again when that happens?